tests/functional/utils/utils.go

```diff
@@ -0,0 +1,38 @@
+package utils
+
+import (
+	"net"
+)
+
+// RandomTCPPort returns an unused TCP port
+// This has a race condition where something else could claim the TCP port
+// before we do
+func RandomTCPPort() (int, error) {
```

Comment:
We could pretty easily add the ability for the back-ends to listen on an ephemeral port and then provide some way to get the port number back, which would eliminate this race condition.

Reply:
Yeah this would be my preferred solution